core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -787,6 +787,14 @@ private[spark] class SparkSubmit extends Logging {
     }
     sparkConf.set(SUBMIT_PYTHON_FILES, formattedPyFiles.split(",").toSeq)
 
+    // set oom error handling in cluster mode
+    if (sparkConf.get(KILL_ON_OOM_ERROR) && deployMode == CLUSTER) {
+      val driverJavaOptions = sparkConf.getOption(SparkLauncher.DRIVER_EXTRA_JAVA_OPTIONS)
+        .map( _ + " ")
+        .getOrElse("") + "-XX:OnOutOfMemoryError=\"kill -9 %p\""
```

Comment:
Since this is a general PR for `SparkSubmit`, does this work on `Windows`?
`ExitOnOutOfMemoryError` will be a better choice, @skonto .

Reply:
Maybe, I lost some context since this is the 3rd try for this.

Reply:
I actually think we should stick w/ `OnOutOfMemoryError`, because `ExitOnOutOfMemoryError` was not present till java 8u92 (discussed earlier here: https://github.com/apache/spark/pull/24796#issuecomment-505063314).  I don't think we specify a minimum version within java 8, so we might have to stick with this.

but yeah, we probably have to make sure it doesn't do anything too weird on windows (does spark actually run in anything other than local mode on windows?)

Reply:
For Windows, I prefer to use JVM option. Personally, I don't think Apache Spark 3.0.0 will be used on JDK 8u91 or older. Apache Spark 3.0.0 starts a new age of JDK11. 😄